ghcide/src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -606,6 +614,36 @@ getCompletions plugins ideOpts CC {allModNamesAsNS, anyQualCompls, unqualCompls,
                   hpos = upperRange position'
               in getCContext lpos pm <|> getCContext hpos pm
 
+
+          -- we need the hieast to be fresh
+          -- not fresh, hasfield won't have a chance. it would to another larger change to ghc IfaceTyCon to contain record fields
```

Comment:
This seems subtle and important: write more! You spent lots of time figuring this out, don't let the knowledge be lost!

Reply:
Also, it's not clear what this comment relates to. Nothing here ensures that the AST is fresh, so what's going on?

Reply:
Made the comment more detailed, and now in proper English. Basically when doing autocomplete its possible for the HieAST to be "stale" meaning its loaded from an hie file rather than generated. But when this happens, it's not possible to get the record fields. This only happens when the current file is first loaded, if an hie file was generated before. As soon as the user edits it, it's fresh again, so I never ran into trouble from this.